astropy/io/fits/tests/test_checksum.py

```diff
@@ -273,11 +273,11 @@ def test_compressed_image_data_float32(self):
             assert 'CHECKSUM' in hdul[1].header
             assert 'DATASUM' in hdul[1].header
 
-            if not sys.platform.startswith('win32'):
-                # The checksum ends up being different on Windows, possibly due
-                # to slight floating point differences
-                assert hdul[1]._header['CHECKSUM'] == 'eATIf3SHe9SHe9SH'
-                assert hdul[1]._header['DATASUM'] == '1277667818'
+            # The checksum ends up being different on Windows and s390/bigendian,
+            # possibly due to slight floating point differences or so?
+            # TODO fix these so they work on all platforms; otherwise pointless.
```

Comment:
Can we still not run them for OSX and little-endian jobs? Maybe @saimn can chime in here?

Reply:
I've raise a separate issue - #10921 - a test that fails on some platforms seems rather useless.

Reply:
Hmm, the fact that it was passing before on travis.org, and that @olebole also says it is passing on the debian CI let me think that this is not an architecture issue. Maybe some floating point differences as the comment says, but it seems weird that a checksum algorithm is not robust to that (and I don't know well enough the algorithm to say anything useful).

Reply:
@mhvk maybe one could just save the resulting fits file and copy it back for debugging?
@saimn IMO checksum [just goes over the bits](https://fits.gsfc.nasa.gov/registry/checksum/checksum) and therefore will fail by any difference.

Reply:
Indeed, we need to debug somehow, but maybe as part of #10921? I don't think it needs to hold up this PR - we should get our CI working again...

Reply:
Maybe add a reference to the new issue here

Reply:
Done